Add error handling and validation to provision-vm.sh

## Problem
Script uses `set -e` but doesn't validate critical operations, leading to silent failures.

**Issues:**
- No validation that Terraform succeeded
- No check that VM IP was obtained
- No verification that cloud-init completed
- Ansible could run against non-existent VM

## Solution
Add validation after each critical step:

```bash
# After terraform apply
if [ $? -ne 0 ]; then
    echo "[ERROR] Terraform failed to create VM" >&2
    exit 1
fi

# Verify VM_IP was set
if [ -z "$VM_IP" ]; then
    echo "[ERROR] Failed to obtain VM IP address" >&2
    echo "Check terraform output and libvirt network" >&2
    exit 1
fi

echo "[SUCCESS] VM created with IP: $VM_IP"

# After cloud-init wait
if \! timeout 300 bash -c "while \! ssh ... 'cloud-init status' 2>/dev/null | grep -q 'status: done'; do sleep 5; done"; then
    echo "[ERROR] cloud-init did not complete within 5 minutes" >&2
    echo "VM may still be initializing. Check with: ssh -i ~/.ssh/vm_key mr@$VM_IP" >&2
    exit 1
fi

# Before ansible
if \! ssh -i ~/.ssh/vm_key -o ConnectTimeout=5 mr@"$VM_IP" 'echo ok' >/dev/null 2>&1; then
    echo "[ERROR] Cannot connect to VM at $VM_IP" >&2
    exit 1
fi
```

## Acceptance Criteria
- [ ] Terraform success validated
- [ ] VM IP verification added
- [ ] Cloud-init timeout check added
- [ ] SSH connectivity verified before Ansible
- [ ] Test with deliberate failures
- [ ] Clear error messages for each failure point

**Estimated time:** 45 minutes
**Reference:** AGENT_REVIEW.md

--- a/provision-vm.sh
+++ b/provision-vm.sh
@@ -112,7 +112,7 @@ echo ""
 echo -e "${YELLOW}Step 2: Waiting for cloud-init to complete...${NC}"
 CLOUD_INIT_SUCCESS=false
 for _ in {1..30}; do
-    if ssh -i ~/.ssh/vm_key -o StrictHostKeyChecking=no -o ConnectTimeout=2 mr@$VM_IP 'cloud-init status --wait' 2>/dev/null; then
+    if ssh -i ~/.ssh/vm_key -o StrictHostKeyChecking=no -o ConnectTimeout=2 mr@"$VM_IP" 'cloud-init status --wait' 2>/dev/null; then
         CLOUD_INIT_SUCCESS=true
         break
     fi

